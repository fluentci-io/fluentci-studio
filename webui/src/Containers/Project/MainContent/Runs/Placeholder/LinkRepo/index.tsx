import LinkRepo from "./LinkRepoWithData";

export default LinkRepo;
