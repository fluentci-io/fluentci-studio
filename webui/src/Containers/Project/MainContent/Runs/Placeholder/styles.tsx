import styled from "@emotion/styled";

export const Container = styled.div`
  padding: 10px;
  color: #fff;
`;

export const Text = styled.span`
  border-radius: 5px;
  padding: 9px;
  background-color: #24f3;
  color: #24ffb5;
  font-size: 13px;
`;

export const Clipboard = styled.button`
  background-color: initial;
  border: none;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
`;
