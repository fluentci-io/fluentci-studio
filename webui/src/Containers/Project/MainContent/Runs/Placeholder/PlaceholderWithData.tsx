import { FC } from "react";
import Placeholder from "./Placeholder";
import { useParams } from "react-router-dom";
import { useGetProjectQuery } from "../../../../../Hooks/GraphQL";

export const PlaceholderWithData: FC = () => {
  const { id } = useParams();
  const { data, loading } = useGetProjectQuery({
    variables: {
      id: id!,
    },
    fetchPolicy: "network-only",
  });
  return <>{!loading && <Placeholder data={data?.project} />}</>;
};

export default PlaceholderWithData;
