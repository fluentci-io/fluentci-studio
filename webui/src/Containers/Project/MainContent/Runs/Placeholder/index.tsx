import Placeholder from "./PlaceholderWithData";

export default Placeholder;
