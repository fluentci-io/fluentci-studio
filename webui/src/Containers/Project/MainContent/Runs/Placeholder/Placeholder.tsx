import { FC } from "react";
import { Container, Text, Clipboard } from "./styles";
import { Project } from "../../../../../Hooks/GraphQL";
import { Terminal } from "@styled-icons/bootstrap";
import { Copy } from "@styled-icons/ionicons-outline";
import copyToClipboard from "copy-to-clipboard";

export type PlaceholderProps = {
  data?: Project | null;
};

const Placeholder: FC<PlaceholderProps> = (props) => {
  const command = import.meta.env.VITE_APP_API_URL?.includes("fluentci.io")
    ? `FLUENTCI_PROJECT_ID=${props.data?.name} fluentci run --remote-exec .`
    : `FLUENTCI_PROJECT_ID=${props.data?.name} fluentci run .`;
  return (
    <Container>
      {props.data?.path !== "empty" && (
        <>
          <div style={{ marginTop: 20 }}>
            <Text>THERE'S NOTHING HERE YET BUT THAT'S OKAY</Text>
          </div>

          <p>Once available, the timeline of pipeline runs will appear here</p>
        </>
      )}
      {props.data?.path === "empty" && (
        <div style={{ display: "flex", padding: 20 }}>
          <div style={{ marginRight: 20 }}>
            <Terminal size={24} />
          </div>
          <div>
            <div style={{ fontWeight: 600 }}>
              Finish setting up your project
            </div>
            <div style={{ fontSize: 14, color: "#7392b1" }}>
              Run the following command in your project directory to finish
              project setup :
              <br />
              <br />
            </div>
            <code
              style={{
                padding: 10,
                backgroundColor: "#5a00e1",
                borderRadius: 3,
              }}
            >
              <span>{command}</span>
              <Clipboard onClick={() => copyToClipboard(command)}>
                <Copy size={20} color="#fff" />
              </Clipboard>
            </code>
          </div>
        </div>
      )}
    </Container>
  );
};

export default Placeholder;
