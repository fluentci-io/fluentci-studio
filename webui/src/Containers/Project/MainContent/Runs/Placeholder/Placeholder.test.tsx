import { render } from "@testing-library/react";
import Placeholder from "./Placeholder";

describe("Placeholder", () => {
  it("should render", () => {
    const { container } = render(<Placeholder />);
    expect(container).toMatchSnapshot();
  });
});
