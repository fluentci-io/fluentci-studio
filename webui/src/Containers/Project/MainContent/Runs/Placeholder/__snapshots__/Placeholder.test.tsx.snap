// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Placeholder > should render 1`] = `
<div>
  <div
    class="css-1pgldjh"
  >
    <div
      style="margin-top: 20px;"
    >
      <span
        class="css-teipoa"
      >
        THERE'S NOTHING HERE YET BUT THAT'S OKAY
      </span>
    </div>
    <p>
      Once available, the timeline of pipeline runs will appear here
    </p>
  </div>
</div>
`;
